### ISSUE TITLE
Support callable for helper functions

### ISSUE THREAD

**Issue by jayS-de** [zordius/lightncandy#228]
It would be really nice to have the support for callables as a helper function. Especially when using lightncandy in a framework you often deal with instances generated by a container. With callable support it would be possible to inject the helper using the framework dependency injection and get rid of the need for static method calls.

**Comment by jayS-de** [comment]
In contrast to the Pull Request I got it working without any change in LightnCandy.

When creating the options array I just add a static call signature like this:

```php
$options['helpers'] = [
  'myHelper' => 'MyHelperClass::helperMethod'
]
```

But the render function is called like this:

```php
$output = $renderer(
    $data,
    [
        'helpers' => [
            'myHelper' => [$myHelperInstance, 'helperMethod']
        ]
    ]
);
```

**Comment by zordius** [comment]
Good! This is a good hint for all.

**Comment by jayS-de** [comment]
Yep. Now I'm able to use the full power of Symfony Dependency Injection, without the hassle of static methods

**Comment by zordius** [comment]
I just pushed dcda1202cf9b75743037fad76364f3169491c422 so we can try this now:

```php
// with FLAG_EXTHELPER we can just provide supported helper names without the helper implementations
$options['helpers'] = [
  'myHelper'
]
```

Or, we can use `$options['helperresolver']` to do helper lookup and just return `1` or `true` when FLAG_EXTHELPER is already enabled.

**Comment by jayS-de** [comment]
Worked perfectly for me with the 

```php
// with FLAG_EXTHELPER we can just provide supported helper names without the helper implementations
$options['helpers'] = [
  'myHelper'
]
```

See also https://github.com/jayS-de/HandlebarsBundle/commit/fcb06b30244e221060ca3df7e26d79deacd4271d

I'm now able to inject helper with symfony dependency injection by using service tags. And i don't need anymore a a nasty hack to inject external helper instances at runtime

**Comment by mAAdhaTTah** [comment]
What about adding support for callable classes implementing `__invoke()`?

**Comment by darkphoenixff4** [comment]
Just to let you know, this doesn't seem to work correctly with block helpers (referring to using FLAG_EXTHELPER).  For whatever reason, although regular helpers are replaced with their contents, block helpers are entirely replaced with '1' (which, of course, is what helperresolver is returning to tell the compiler that the helper will be provided later).

**Comment by theodorejb** [comment]
@jenschude Check out [PHP Handlebars](https://github.com/devtheorem/php-handlebars), which has better performance and support for the full Handlebars spec.

In PHP Handlebars helpers are always passed at runtime, just like in Handlebars.js, and as long as each helper is a closure it will work as expected. For example, anonymous functions can explicitly `use` data from the current scope, or a helper can be an arrow function, or even a first-class callable reference to a built-in function. For example:

```php
use DevTheorem\Handlebars\Handlebars;

require 'vendor/autoload.php';

$templateStr = '{{dateFormat (getDate dateStr) "M j, Y"}}';

$helpers = [
    'dateFormat' => date_format(...),
    'getDate' => fn(string $date) => new DateTimeImmutable($date),
];

$template = Handlebars::compile($templateStr);

$data = ['dateStr' => '2026-03-30'];

echo $template($data, ['helpers' => $helpers]);
```
Output: `Mar 30, 2026`
